src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,443 @@
+            str(aggr_name) +
+            '|" ' +
+            file_path)
+        cmds.append(
+            'sed -i "s|aggregate_metadata:.*|aggregate_metadata: ' +
+            str(aggr_metadata) +
+            '|" ' +
+            file_path)
+        cmds.append(
+            'sed -i "s|hosts:.*|hosts: ' +
+            str(hostname_list) +
+            '|" ' +
+            file_path)
+        for cmd in cmds:
+            status = os.system(cmd)
+            print "cmd: {}".format(cmd)
+            if status != 0:
+                raise Exception(
+                    "Failed to execute the command {}"
+                    " with error code {}".format(
+                        cmd, status))
+
+    def create_aggregates(self, aggr_name):
+        env_opts = \
+            " -e ~/pilot/templates/create_aggregate_environment.yaml"
+
+        cmd = self.overcloudrc + "openstack stack create " \
+            " {}" \
+            " --template" \
+            " ~/pilot/templates/overcloud/puppet/" \
+            "services/dellnfv/createaggregate.yaml" \
+            " {}" \
+            "".format(aggr_name,
+                      env_opts)
+        aggregate_create_status = os.system(cmd)
+        if aggregate_create_status == 0:
+            logger.info("Aggregate {} created".format(aggr_name))
+        else:
+            logger.error(
+                "Aggregate {} could not be created..."
+                " Exiting post deployment tasks".format(aggr_name))
+            sys.exit(0)
+
+    def set_aggregate_metadata(self):
+        try:
+                # Get the overcloud details
+            ConfigOvercloud.get_overcloud_details(self.overcloud_name)
+
+            # Create nova client object
+            nova = nvclient.Client(
+                2,
+                UC_USERNAME,
+                UC_PASSWORD,
+                UC_PROJECT_ID,
+                UC_AUTH_URL)
+            aggregate_dict = dict()
+            hostname_list = self.get_dell_compute_nodes_hostnames(nova)
+
+            aggregate_dict.update(DELL_AGGREGATE_METADATA)
+
+            for aggregate_name, aggregate_metadata in aggregate_dict.items():
+                self.edit_aggregate_environment_file(
+                    aggregate_name, aggregate_metadata, hostname_list)
+                self.create_aggregates(aggregate_name)
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            logger.error(
+                "Unable to set aggregate metadata."
+                " Exiting post deployment task...")
+            sys.exit(0)
+
+    def post_deployment_tasks(self, enable_hugepages, enable_numa, hpg_size):
+        try:
+            # Reboot all the Dell NFV compute nodes
+            self.reboot_dell_nfv_nodes()
+
+            # Wait for 5 mins to let dll nfv nodes boot up
+            time.sleep(300)
```

Comment:
Please dont sleep .. but rather check the nodes are coming back up , should have some samples of similar code already on master.

Reply:
Fixed